### DISCUSSION TITLE
populate not work with OR filter

### DISCUSSION THREAD

**Discussion by 47vigen** [mikro-orm/mikro-orm#2618] [Q&A] [answered]
**Describe the bug**
I have an issue with em.find() when i use where argument and filter with "OR" !
I pass this arguments to em.find() 
```
  "where": { "$or": [{ "author": { "email": { "$eq": "47vigen@gmail.com" } } }, { "title": { "$ilike": "%snake%" } }] },
  "options": { "limit": 20, "offset": 0, "cache": 3600000, "populate": ["author"] }
```
but I did not get author!

**To Reproduce**
1. create article entity 
2. create user entity
3. many to one relation : articles(User) <=> author(Article)
4. run em.find(Article, { $or: [{ author: { email: { $eq: '47vigen@gmail.com' } } }, { title: { $ilike: '%snake%' } }] }, { limit: 20, offset: 0, cache: 3600000, populate: ['author'] })
5. Articles are taken but without its authors

**Expected behavior**
Because I added an "author" in the populate field, I expect articles to be received with its authors.

**Additional context**
you can test graphql online demo on https://venins.ir/graphiql with this query :
``
{
  articles(
    filters: {or: [{author: {email: {eq: "47vigen@gmail.com"}}}, {title: {ilike: "%snake%"}}]}
  ) {
    data {
      id
      slug
      title
      author {
        id
        createdAt
      }
    }
  }
}
``

**Versions**

| Dependency | Version |
| - | - |
| node | 14.17.5 |
| typescript | ✔ |
| mikro-orm | 4.5.10 |
| your-driver | postgresql |

**Comment by B4nan** [comment, marked as answer]
Please share some code if you want help. I have no idea what your app is doing and how is it written. Note that reproductions need to be minimal - only ORM should be involved, no GraphQL, no webservers - if you want to report bugs in ORM, you should use only the ORM to reproduce. I would also expect you to at least enable debug mode and compare what queries the ORM is using and say what you want instead.

edit: oh now I get it, you are the guy with GQL example PR. anyway, bug reports need to be minimal as said above :] and I am quite sure the suggestion underneath is what you need to do to fix it

One problem you might be facing is that query conditions are propagated to populate queries. This means that your author relations will be constrained to the query too, which might not be what you want. If you want to populate all authors, do it separately:

```ts
const articles = await em.find(Article, { $or: [{ author: { email: { $eq: '47vigen@gmail.com' } } }, { title: { $ilike: '%snake%' } }] }, { limit: 20, offset: 0, cache: 3600000 });
await em.populate(articles, ['author']);
```

Otherwise if you just use `FindOptions.populate`, the query for authors will contain the `{ author: { email: { $eq: '47vigen@gmail.com' } } }` part too.

**Reply by 47vigen** [reply]
thank for your answer I use your suggestion and its work!
but another problem about lazy property :)
I put lazy properties in em.populate(data, populate) but no trace of them in the returned data

**Reply by B4nan** [reply]
That works only in v5

**Reply by 47vigen** [reply]
well Everything works fine!
Thank you very much for your help
I think the example I created could be creative and helpful, probably because of the filterable queries it has.
Thanks if you check this example and maybe merge it 💚
